### DISCUSSION TITLE
Trying to implement multi database setup like vaultwarden did

### DISCUSSION THREAD

**Discussion by SamTV12345** [dani-garcia/vaultwarden#3421] [Q&A] [answered]
I am trying to rebuild my [project](https://github.com/SamTV12345/PodFetch.git) on the branch feature/multiple-databases with multi database support like sqlite, postgres and mysql. I therefore took the relevant pieces from this repository and tried to puzzle them together but I am failing and get a lot of error messages in my diesel statements.

The entities worked before but now I am getting the following errors:
- trait bound `Podcast: FromSqlRow<_, _>` is not satisfied
- the trait bound `(diesel::sql_types::Integer, diesel::sql_types::Text, diesel::sql_types::Text, diesel::sql_types::Text, diesel::sql_types::Text, diesel::sql_types::Integer, diesel::sql_types::Nullable<diesel::sql_types::Text>, diesel::sql_types::Nullable<diesel::sql_types::Text>, diesel::sql_types::Nullable<diesel::sql_types::Text>, diesel::sql_types::Nullable<diesel::sql_types::Text>, diesel::sql_types::Nullable<diesel::sql_types::Text>, diesel::sql_types::Nullable<diesel::sql_types::Text>, diesel::sql_types::Bool, diesel::sql_types::Text): load_dsl::private::CompatibleType<Podcast, _>` is not satisfied
etc. Could someone please help me in getting this to work?

**Comment by stefan0xC** [comment]
> I therefore took the relevant pieces from this repository

Just a quick note: Vaultwarden is [licensed under the terms of the AGPLv3](https://github.com/dani-garcia/vaultwarden/blob/main/LICENSE.txt), so if you took something you have to grant your users the same [_freedoms_](https://www.gnu.org/philosophy/free-sw.en.html) because the terms of [your projects current license](https://github.com/SamTV12345/PodFetch/blob/main/LICENSE) are too _permissive_.

**Reply by SamTV12345** [reply, marked as answer]
Thanks for the reminder. I'm wondering why a project would want to be AGPLv3 but I guess I'm forced to use this license too once I get this merged in main.
